### PR TITLE
Streamlines the ghost-to-player creation process

### PR DESCRIPTION
**Pull request by Crazylemon64**
:cl:Crazylemon
rscadd: Adds an "ethereal beacon", an admin-only structure usable to attract ghosts to an area.
rscadd: Adds several transformer machines to streamline the ghost-to-PC process.
/:cl:

The "Ethereal beacon" can be activated by hand by an admin, to periodically (normally every 30 seconds) call ghosts to where the beacon is. This can be adjusted by modifying `ghost_alert_delay`, which is in deciseconds.
The "Transmogrifier" has a `target_species` var, and turns any humanoid that passes through it into the corresponding species.
The "Auto-equipper 9000" has a `selected_outfit` var, which is a path to an outfit datum to apply to any human that passes through. You may also want to set `transform_standing` to 1 to let outfits that put items in people's hands do their thing.
The "DNA scrambler" scrambles the appearance of anyone who enters it, to "sanitize" the mob of being any other character.
The "genetic blueprint applier" uses a genetic sample from a gene disk to make all humans that pass through it a carbon-copy of the given DNA. Useful for literal clone wars. Has a `locked` variable if you don't want people tampering with it.

Also refactors ERT equipment code to work off of outfit datums, so we can use ERT outfits with the machines above.

### CODE REVIEW

**Comment by Anasari** [comment]
The clone wars begin

File: code/game/response_team.dm [hunk 1]
```diff
@@ -345,242 +271,344 @@ var/ert_request_answered = 0
...
-			M.equip_to_slot_or_del(new /obj/item/weapon/restraints/handcuffs(M), slot_in_backpack)
-			M.equip_to_slot_or_del(new /obj/item/weapon/storage/lockbox/loyalty(M), slot_in_backpack)
-			M.equip_to_slot_or_del(new /obj/item/weapon/gun/energy/pulse/pistol(M), slot_in_backpack)
-
-			M.equip_to_slot_or_del(new /obj/item/weapon/pinpointer(M), slot_l_store)
-			M.equip_to_slot_or_del(new /obj/item/weapon/melee/classic_baton/telescopic(M), slot_r_store)
-
+/datum/outfit/job/centcom/response_team
+	name = "Response team"
+	var/RTassignment = "Emergency Response Team Member"
```
Comment (tigercat2000) [inline code/game/response_team.dm]: please use lower snake case for these variables, literally everything else that uses outfit datums does so

Comment (Crazylemon64) [inline code/game/response_team.dm]: Alright

**Comment by Fruerlund** [comment]
Nice tool for events!

**Comment by Fruerlund** [comment]
May i suggest in regards to datums. At the moment when turning a mobs appearance and outfits into a datum everything saves accordingly. 

However if I were to var edit icon_state and item_color these things do not carry over which is sad as they allow you to come up with many new outfits guns items and so on

**Comment by pinatacolada** [comment]
op pls nerf

**Comment by SamHPurp** [comment]
Ghost bar precursor? 👻

**Comment by scrubmcnoob** [comment]
woo event tools.
Thanks man.

File: code/game/objects/structures/misc.dm [hunk 2]
```diff
@@ -72,4 +72,46 @@
...
+/obj/structure/ghost_beacon/initialize()
+	. = ..()
+	last_ghost_alert = world.time
+	if(active)
+		processing_objects.Add(src)
+
+/obj/structure/ghost_beacon/Destroy()
+	if(active)
+		processing_objects.Remove(src)
+	. = ..()
```
Comment (Fox-McCloud) [inline code/game/objects/structures/misc.dm]: I know this is the same thing as `return ..()`, but it's a bit inconsistent to use this at this location.

**Comment by fludd12** [comment]
Significantly better than my responder machine, thank you for this!

**Pull request merged**